payments/webhooks.py

```diff
@@ -189,6 +192,36 @@ def handle_subscription_cancelled(cls, uid: str):
             external_id=None,
         )
 
+    @classmethod
+    def handle_invoice_failed(cls, uid: str, data: dict):
+        logger.info(f"Invoice failed: {data}")
+
+        if stripe.Charge.list(payment_intent=data["payment_intent"], limit=1).has_more:
+            # we must have already sent an invoice for this to the user. so we should just ignore this event
+            logger.info("Charge already exists for this payment intent")
+            return
+
+        if data.get("metadata", {}).get("auto_recharge"):
+            logger.info("auto recharge failed... sending invoice email")
+            send_payment_failed_email_with_invoice.delay(
+                uid=uid,
+                invoice_url=data["hosted_invoice_url"],
+                dollar_amt=data["amount_due"] / 100,
+                subject="Payment failure on your Gooey.AI auto-recharge",
+            )
+        elif data.get("subscription_details", {}):
+            print("subscription failed")
+            send_payment_failed_email_with_invoice.delay(
+                uid=uid,
+                invoice_url=data["hosted_invoice_url"],
+                dollar_amt=data["amount_due"] / 100,
+                subject="Payment failure on your Gooey.AI subscription",
+            )
+        else:
+            print("not auto recharge or subscription")
```

Comment:
print on sigle line and use loguru if loggin in prod

Reply:
grr, this shouldn't have been committed